aten/src/ATen/native/cuda/LinearAlgebraStubs.cpp

```diff
@@ -0,0 +1,188 @@
+  lu_solve_stub(DeviceType::CUDA, b, lu, pivots);
+}
+
+void lazy_lstsq_kernel(const Tensor& a, Tensor& b, Tensor& rank, Tensor& singular_values, Tensor& infos, double rcond, std::string driver_name)  {
+  getTorchLinalgLibrary();
+  lstsq_stub(DeviceType::CUDA, a, b, rank, singular_values, infos, rcond, driver_name);
+}
+
+REGISTER_CUDA_DISPATCH(cholesky_stub, &lazy_cholesky_kernel)
+REGISTER_CUDA_DISPATCH(cholesky_inverse_stub, &lazy_cholesky_inverse_kernel);
+REGISTER_CUDA_DISPATCH(lu_factor_stub, &lazy_lu_factor);
+REGISTER_CUDA_DISPATCH(triangular_solve_stub, &lazy_triangular_solve_kernel);
+REGISTER_CUDA_DISPATCH(orgqr_stub, &lazy_orgqr_kernel);
+REGISTER_CUDA_DISPATCH(ormqr_stub, &lazy_ormqr_kernel);
+REGISTER_CUDA_DISPATCH(geqrf_stub, &lazy_geqrf_kernel);
+REGISTER_CUDA_DISPATCH(linalg_eigh_stub, &lazy_linalg_eigh_kernel);
+REGISTER_CUDA_DISPATCH(eig_stub, &lazy_eig_kernel);
+REGISTER_CUDA_DISPATCH(linalg_eig_stub, &lazy_linalg_eig_kernel);
+REGISTER_CUDA_DISPATCH(svd_stub, &lazy_svd_kernel)
+REGISTER_CUDA_DISPATCH(lu_solve_trans_stub, &lazy_lu_solve_trans);
+REGISTER_CUDA_DISPATCH(lu_solve_stub, &lazy_lu_solve);
+REGISTER_CUDA_DISPATCH(lstsq_stub, &lazy_lstsq_kernel);
+} // anonymous namespace
+
+// Old style dispatch
+Tensor& _linalg_inv_out_helper_cuda(Tensor &result, Tensor& infos_lu, Tensor& infos_getri) {
+    static auto impl = getTorchLinalgFunc<decltype(&_linalg_inv_out_helper_cuda)>("_ZN2at6native27_linalg_inv_out_helper_cudaERNS_6TensorES2_S2_");
+    TORCH_CHECK(impl != nullptr, "Can't find _linalg_inv_out_helper_cuda");
+    return impl(result, infos_lu, infos_getri);
+}
+
+std::tuple<Tensor, Tensor> legacy_lstsq_cuda(const Tensor &B, const Tensor &A) {
+    static auto impl = getTorchLinalgFunc<decltype(&legacy_lstsq_cuda)>("_ZN2at6native17legacy_lstsq_cudaERKNS_6TensorES3_");
+    TORCH_CHECK(impl != nullptr, "Can't find legacy_lstsq_cuda");
+    return impl(B, A);
+}
+
+Tensor _cholesky_solve_helper_cuda(const Tensor& self, const Tensor& A, bool upper) {
+    static auto impl = getTorchLinalgFunc<decltype(&_cholesky_solve_helper_cuda)>("_ZN2at6native27_cholesky_solve_helper_cudaERKNS_6TensorES3_b");
+    return impl(self, A, upper);
+}
+
+std::tuple<Tensor, Tensor> _linalg_qr_helper_cuda(const Tensor& input, c10::string_view mode) {
+    static auto impl = getTorchLinalgFunc<decltype(&_linalg_qr_helper_cuda)>("_ZN2at6native22_linalg_qr_helper_cudaERKNS_6TensorEN3c1017basic_string_viewIcEE");
+    return impl(input, mode);
+}
+
+std::tuple<Tensor, Tensor> _symeig_helper_cuda(const Tensor& self, bool eigenvectors, bool upper) {
+    static auto impl = getTorchLinalgFunc<decltype(&_symeig_helper_cuda)>("_ZN2at6native19_symeig_helper_cudaERKNS_6TensorEbb");
+    return impl(self, eigenvectors, upper);
+}
+
+std::tuple<Tensor, Tensor> _solve_helper_cuda(const Tensor& self, const Tensor& A) {
+    static auto impl = getTorchLinalgFunc<decltype(&_solve_helper_cuda)>("_ZN2at6native18_solve_helper_cudaERKNS_6TensorES3_");
```

Comment:
Would it be simpler to have a `Hooks` class similar to `CUDAHooks` or `VariableHooks`. Then this would just looks like:
```
getTorchLinalgLibrary()->_solve_helper_cuda(self, A);
```